subaccounts.go

```diff
@@ -274,27 +249,17 @@ func (c *Client) SubaccountContext(ctx context.Context, id int) (subaccount *Sub
 			slist := map[string]Subaccount{}
 			err = json.Unmarshal(body, &slist)
 			if err != nil {
-				return
```

Comment:
Line 243 above should be removed. It can never be anything except 200 ok.

Also I am not certain this logic is correct. It seems too restrictive. Isn't is OK as long as any HTTP 2xx response is returned?